lib/push_2_tweet_communication.js

```diff
@@ -0,0 +1,156 @@
+    return process.nextTick(callback);
+  } else {
+    var err = new caErrors.BadPayload(
+      request.payload
+    );
+    err.message += ' (errors: ' + p2tHelper.stringifyRevalidatorErrors(validation.errors) + ')';
+    p2tLogger.warn(request.push2Tweet.context, 'The request payload: ' + request.payload + ' is NOT valid' +
+      (err.message ? ', error=' + JSON.stringify(err) : '')
+    );
+    return process.nextTick(callback.bind(null, err));
+  }
+}
+
+/**
+ * Processes and returns the requested operation information
+ * @param {object} request The updateContext request received
+ * @param {function} callback Callback to be called with the result of the operation
+ */
+function getOperationDescriptor(request, callback) {
+  checkRequest(request, function(err) {
+    if (err) {
+      return process.nextTick(callback.bind(null, err));
+    } else {
+      var payloadJSON = JSON.parse(request.payload);
+      var operationDescriptor = {
+        button: payloadJSON.button,
+        action: payloadJSON.action,
+        extra: payloadJSON.extra,
+        callback: payloadJSON.callback
+      };
+      request.push2Tweet.operationDescriptor = operationDescriptor;
+      return process.nextTick(callback);
+    }
+  });
+}
+
+/**
+ * Responds to the received request
+ * @param err Error in case an error ocurred during the processing
+ * @param request The received request
+ * @param reply hapi's reply() function
+ * @param callback A callback to notify whenthe operation has finished
+ * @return {*} This function does not return anything of value
+ */
+function respond2Request(err, request, reply, callback) {
+  var response = {
+    externalId: Math.floor(Math.random() * (100 - 1 + 1)) + 1,
+    buttonId: (request && request.push2Tweet && request.push2Tweet.operationDescriptor &&
+      request.push2Tweet.operationDescriptor.button) || 'unknown'
+  };
+  if (err) {
+    response.details = {
+      code: err.code || Array.isArray(err) && err[0].code || 'ERROR',
+      message: err.message || Array.isArray(err) && err[0].message || 'Some error occured when sending the tweet'
+    };
+    p2tLogger.warn(request.push2Tweet.context, 'Responding to the request ' + request.method.toUpperCase() + ' ' +
+      request.url.path + ' and payload: ' + request.payload +
+      ' with the following response: ' + JSON.stringify(response)
+    );
+    reply(response);
+  } else {
+    response.details = {
+      code: request.push2Tweet.result.code,
+      tweet: request.push2Tweet.result.tweet
+    };
+    p2tLogger.debug(request.push2Tweet.context, 'Responding to the request ' + request.method.toUpperCase() + ' ' +
+      request.url.path + ' and payload: ' + request.payload +
+      ' with the following response: ' + JSON.stringify(response)
+    );
+    if (request.push2Tweet.operationDescriptor.callback) {
+      reply();
+      setTimeout(function() {
```

Comment:
Fixed in https://github.com/telefonicaid/push-2-tweet/commit/d5650d46a92509aeda258bd876bbc95e2c5c0dc3